### PR TITLE
Feature/validatelogin

### PR DESCRIPTION
**Pull request by nicosammito**

### CODE REVIEW

File: class/User.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     * validates the email and password and returns a JSON object with a jwt token which is needed to auth the user
+     * and a boolean
+     * @param {String} email
+     * @param {String} password
+     * @param {String} token
+     * @returns {Promise<void>}
+     */
+    async validateLogin(email, password, token) {
+        const response = await fetch(BASE_URL+USER_ENDPOINTS.validate_login, {
+            method: 'GET',
```
Comment (Taucher2003) [inline class/User.js:19]: Why in the world would you use GET for that?

File: class/User.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                email: email,
+                password: password,
+                token: token
```
Comment (Taucher2003) [inline class/User.js:26]: You don't put the token in the body. That will always give you 400 Bad Request.

File: Utils/constants.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+
+const BASE_URL = "https://api.hamibot.io/PREALPHA/";
```
Comment (Taucher2003) [inline Utils/constants.js:3]: Normally an URL is lowercase

File: Utils/constants.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+const USER_ENDPOINTS = {
+    validate_login : "/users/login"
+}
+
+export{BASE_URL, USER_ENDPOINTS};
```
Comment (Taucher2003) [inline Utils/constants.js:9]: That should be possible but we always used module.exports

File: class/User.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import {BASE_URL, USER_ENDPOINTS} from "../Utils/constants";
```
Comment (Taucher2003) [inline class/User.js:1]: As de normally use module.exports we also sie require() instead of import

**Review by Taucher2003 — CHANGES_REQUESTED** [review]
There are some things that can work and just don't follow the way how we did it normally.
But also there are things that wouldn't work at all.

**Review by Th3Ph4nt0m — CHANGES_REQUESTED** [review]
The RESTAPI is not finished yet. It is useless to start this API here, because a lot of changes will still take place in the REST.

File: class/User.js (hunk 5 above)
Comment (nicosammito) [reply class/User.js:1]: this is not node its only Javascript so require don't exist

File: class/User.js (hunk 2 above)
Comment (nicosammito) [reply class/User.js:26]: so in the header right?

File: class/User.js (hunk 1 above)
Comment (nicosammito) [reply class/User.js:19]: you don't post something you just want to get

File: Utils/constants.js (hunk 4 above)
Comment (nicosammito) [reply Utils/constants.js:9]: its not possible because its not node

File: Utils/constants.js (hunk 3 above)
Comment (nicosammito) [reply Utils/constants.js:3]: but const in js should always be uppercase

File: class/User.js (hunk 1 above)
Comment (Taucher2003) [reply class/User.js:19]: You need POST because that is defined in the backend.

File: class/User.js (hunk 2 above)
Comment (Taucher2003) [reply class/User.js:26]: No, you don't have a token while logging in. You get the token as response.

File: Utils/constants.js (hunk 3 above)
Comment (Taucher2003) [reply Utils/constants.js:3]: I meant the version, not the variable name.

File: Utils/constants.js (hunk 4 above)
Comment (Taucher2003) [reply Utils/constants.js:9]: Ok, didn't know that. My bad

File: class/User.js (hunk 5 above)
Comment (Taucher2003) [reply class/User.js:1]: Same like above. Sorry

**Comment by Th3Ph4nt0m** [comment]
This API will not be implemented before the REST API has at least a pre-release. Conversation locked.

**Pull request closed without merging**